src/gflownet/data/sampling_iterator.py

```diff
@@ -59,6 +64,9 @@ def _idx_iterator(self):
             # Otherwise, figure out which indices correspond to this worker
             worker_info = torch.utils.data.get_worker_info()
             n = len(self.data)
+            if n == 0:
```

Comment:
Since this function is a generator, I think of the `return` statements as `break`. As for why `yield` rather than constructing a list and returning it, I guess I thought it would be more elegant this way. `yield` handles the case where `stream == True` and we'd have to return an infinite iterator anyways.